src/Definition/HTMLDefinition.hack

```diff
@@ -1916,12 +1916,12 @@ class HTMLPurifier_HTMLDefinition extends HTMLPurifier\HTMLPurifier_Definition {
 			C\is_empty($allowed_attributes)
 		) {
 			$allowed = (string)$config->def->defaults['HTML.Allowed'];
-			if ($allowed !== '') {
+			if ($allowed is string && $allowed !== '') {
 				list($allowed_elements, $allowed_attributes) = $this->parseTinyMCEAllowedList($allowed);
 			}
 		}
 
-		if ($allowed_elements is dict<_, _> && !C\is_empty($allowed_elements)) {
```

Comment:
Yea I think the reason why we never caught this in the first place is because of the way the PHP purifier got _everything_ from the serializer. I'd be down to do some more investigation in another ticket though